File: docs/code_examples/text_log.rs
//! Log a `TextLog`

use rerun::{archetypes::TextLog, RecordingStreamBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_text_log").memory()?;

    rec.log(
        "log",
        &TextLog::new("Application started.").with_level("INFO"),
    )?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/snippets/all/archetypes/text_log.cpp
// Log a `TextLog`

#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_text_log");
    rec.spawn().exit_on_failure();

    rec.log("log", rerun::TextLog("Application started.").with_level(rerun::TextLogLevel::Info));
}
